Add outbreak distribution vignette (#304)
epiverse-trace/epichains: vignettes/outbreak_distribution.Rmd

---

> title: "Outbreak size and length distribution"

Reviewer: ```suggestion
title: "Simulating scenarios of outbreak size and length"
```

The factors that influence outbreak size are the reproduction number of the infectious disease, the heterogeneity in transmission dynamics, population susceptibility, interventions and stochasticity in the epidemic.

Using a branching process model we can seed a new epidemic with a single index case, and simulate the epidemic by defining an _offspring distribution_, which specifies the probability that each infected individual will go on to infect a further _N_ individuals (\( P(n = N) = p_N, N \in \{0, 1, 2, \ldots\} \)). The branching process simulation model can sample from the offspring distribution to generate a transmission chain, from which the outbreak size or outbreak length can be calculated. The branching process model is stochastic due to the random sampling of offspring distribution.

::: {.alert .alert-primary}
**Glossary**

* Outbreak _size_: the total number of cases produced by an outbreak before it goes extinct.

* Outbreak _length_: the total (maximum) number of generations reached by a outbreak before it goes extinct.
:::

> In this vignette we will outline how to show the distribution of outbreak sizes and lengths using the _epichains_ R package. 

Reviewer: ```suggestion
In this vignette we will outline how to generate a distribution of outbreak sizes and lengths using the _epichains_ R package. 
```

## Simulating an outbreak size distribution

First, we'll simulate the outbreak size distribution using the a Poisson offspring distribution. This is a simple one parameter distribution where the mean and variance of the distribution defined by $\lambda$ which is equal to the reproduction number ($R$).

For this example we'll vary the reproduction number between 0.1 and 1.0, at 0.1 intervals.

```{r, define-parameters}
statistic <- "size"
offspring_dist <- "rpois"
R <- seq(0.1, 1.0, 0.1)
```

We will create a parameter space will each combination of parameters. In this simple outbreak size simulation, only the reproduction number varies. 

```{r, parameter-grid}
scenarios <- expand.grid(
  offspring_dist = offspring_dist,
  statistic = statistic,
  R = R,
  stringsAsFactors = FALSE
)
```

> The branching process is stochastic so we need to specify how many simulation model replicates to run to capture the stochasticity in the epidemic. This is important, as mentioned above as a factor that influences outbreak size, the same epidemic transmissibility parameters can result in different sized outbreaks depending on randomness, such as the number of infectees from a given infector.

Reviewer: Branching processes are stochastic processes, so we need to specify how many simulation model replicates to run to capture the variability in the epidemic. This is important, as mentioned above as a factor that influences outbreak size, the same epidemic transmissibility parameters can result in different sized outbreaks depending on randomness, such as the number of infectees from a given infector.
```

```{r, define-n}
n_chains <- 1000
```

Here we choose to simulate each scenario `r n_chains` times.

```{r, define-breaks}
breaks <- c(0, 2, 5, 10, 20, 50, 100, Inf)
```

> 

Reviewer: This needs a leading statement.

```{r, outbreak-sim-pois-size}
outbreak_list <- vector(mode = "list", length = nrow(scenarios))
for (i in seq_len(nrow(scenarios))) {
  offspring_dist_fun <- match.fun(scenarios[i, "offspring_dist"])

>   outbreak_list[[i]] <- epichains::simulate_chain_stats(

Reviewer: ```suggestion
  outbreak_list[[i]] <- simulate_chain_stats(
```

  ) +
  ggplot2::scale_x_discrete(name = "Reproduction number (R)") +
  ggplot2::scale_y_continuous(name = "Proportion of outbreaks") +
  ggplot2::scale_fill_brewer(
    name = "Outbreak size", 
    palette = "Spectral"
  ) + 
  ggplot2::theme_bw()
```

```{r, min-outbreak-size, echo=FALSE}
max_R <- outbreak_size[outbreak_size$R == max(outbreak_size$R), ]
freq <- max_R$Freq[which(max_R$interval == "(0,2]")]
```

The plot shows that the majority of infections do not seed an outbreak, as most index cases produce epidemics with less than two infections, including the index case. This is true even for the largest reproduction number used ($R$ = `r max(R)`), where `r freq * 100`% have fewer or equal to two total infections.

As the reproduction number increases towards 1, the possibility of an outbreak occurring with more than 20 cases becomes more probable.

## Simulating an outbreak length distribution

Instead of looking at the outbreak size distribution, i.e. the total number of cases in an outbreak, we may instead be interested in the outbreak _length_ distribution. This is the distribution of the number of generations that an epidemic reaches before going extinct.

> We can follow the same procedure as for the outbreak size, but specify `stasitics = "length` in the `simulate_chain_stats()` function. We use the same parameters as the outbreak size simulation above, simulating with a Poisson offspring distribution.

Reviewer: ```suggestion
We can follow the same procedure as for the outbreak size, but specify `statistic = "length` in the `simulate_chain_stats()` function. We use the same parameters as the outbreak size simulation above, simulating with a Poisson offspring distribution.
```

    labeller = ggplot2::label_both
  ) +
  ggplot2::theme_bw()
```

The increased heterogeneity in epidemic transmission leads to small expected outbreak sizes. The upper-left plot (smallest $k$) shows the fewest outbreak clusters with more than 20 cases. This results from the mean number of secondary infections per case ($R$) is influenced by a few individuals that disproportionally infect many individuals, but the majority of primary cases infect few, if any, individuals. However, in those few replicates where the _superspreaders_ infect many individuals the epidemic can become very large (i.e. $ >20$ cases). 

Where $k$ is `r k[2]`, `r k[3]`, and `r k[4]` there is little difference in the outbreak size distributions. As the values of $k$ gets very large (i.e. $ >1000$) the outbreak size distribution from a Negative Binomial offspring distribution converges on the outbreak size distribution from a Poisson offspring distribution, with an equal $R$. 

We do not repeat the example of simulating an outbreak length distribution using a Negative binomial distribution, but this can be achieved using the code from the examples above, setting `statistic = "length"`.

> ## Simulating an outbreak size distribution excluding the index case

Reviewer: I think this section doesn't need to be long as it's quite intuitive. It can be a sentence or two explaining the use case and how it can be done in code, i.e., just subtract 1 case.
